src/org/javarosa/xpath/expr/XPathEqExpr.java

```diff
@@ -25,71 +25,77 @@
 import org.javarosa.core.util.externalizable.DeserializationException;
 import org.javarosa.core.util.externalizable.ExtUtil;
 import org.javarosa.core.util.externalizable.PrototypeFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.lang.Math.abs;
+import static org.javarosa.xpath.expr.XPathFuncExpr.toBoolean;
+import static org.javarosa.xpath.expr.XPathFuncExpr.toNumeric;
+import static org.javarosa.xpath.expr.XPathFuncExpr.unpack;
 
 public class XPathEqExpr extends XPathBinaryOpExpr {
+    private static final Logger logger = LoggerFactory.getLogger(XPathEqExpr.class.getSimpleName());
     public boolean equal;
 
-    public XPathEqExpr () { } //for deserialization
+    /** Deserialization constructor */
+    public XPathEqExpr() {
+        logger.debug("XPathEqExpr{}()", id(), equal, a, b);
```

Comment:
Does this only log the first of the substitution params logged?

Reply:
Thanks for catching that. I’ll don’t think that’s what I intended.

Reply:
That’s fixed, I ran tests and saw that the output from that was unchanged, rebased on master, and force pushed.